interfaces/tests/sync_int/sync_int.c

```diff
@@ -0,0 +1,211 @@
+    wait_on_address(ptr, &value, UINT32_MAX);
+    interlocked_increment(ptr);
+    ThreadAPI_Exit(0);
+    return 0;
+}
+
+
+BEGIN_TEST_SUITE(sync_int)
+
+
+TEST_SUITE_INITIALIZE(a)
+{
+    g_testByTest = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(g_testByTest);
+}
+
+TEST_SUITE_CLEANUP(b)
+{
+
+    TEST_MUTEX_DESTROY(g_testByTest);
+}
+
+TEST_FUNCTION_INITIALIZE(c)
+{
+    if (TEST_MUTEX_ACQUIRE(g_testByTest))
+    {
+        ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework");
+    }
+}
+
+TEST_FUNCTION_CLEANUP(d)
+{
+    TEST_MUTEX_RELEASE(g_testByTest);
+}
+
+/*Codes_SRS_SYNC_43_001: [ wait_on_address shall atomically compare *address and *compare_address.]*/
+/*Codes_SRS_SYNC_43_007: [ If *address is equal to *compare_address, wait_on_address shall cause the thread to sleep. ]*/
+/*Codes_SRS_SYNC_43_008: [wait_on_address shall wait indefinitely until it is woken up by a call to wake_by_address_[single/all] if timeout_ms is equal to UINT32_MAX]*/
+/*Codes_SRS_SYNC_43_003: [ wait_on_address shall wait until another thread in the same process signals at address using wake_by_address_[single/all] and return true. ]*/
+TEST_FUNCTION(two_threads_increment_alternately)
+{
+    ///arrange
+    volatile_atomic int32_t var;
+    interlocked_exchange(&var, 0);
+
+    THREAD_HANDLE thread1;
+    THREAD_HANDLE thread2;
+    ///act
+    ASSERT_ARE_EQUAL(THREADAPI_RESULT, THREADAPI_OK, ThreadAPI_Create(&thread1, increment_on_even_values, (void*)&var));
+    ASSERT_ARE_EQUAL(THREADAPI_RESULT, THREADAPI_OK, ThreadAPI_Create(&thread2, increment_on_odd_values, (void*)&var));
+
+
+
+    ///assert
+    int return1;
+    ThreadAPI_Join(thread1, &return1);
+    int return2;
+    ThreadAPI_Join(thread2, &return2);
+    ASSERT_ARE_EQUAL(int32_t, 99, interlocked_add(&var, 0), "Return value is incorrect");
+}
+
+/*Codes_SRS_SYNC_43_001: [ wait_on_address shall atomically compare *address and *compare_address.]*/
+/*Codes_SRS_SYNC_43_007: [ If *address is equal to *compare_address, wait_on_address shall cause the thread to sleep. ]*/
+/*Codes_SRS_SYNC_43_008: [wait_on_address shall wait indefinitely until it is woken up by a call to wake_by_address_[single/all] if timeout_ms is equal to UINT32_MAX]*/
+/*Codes_SRS_SYNC_43_003: [ wait_on_address shall wait until another thread in the same process signals at address using wake_by_address_[single/all] and return true. ]*/
+/*Codes_SRS_SYNC_43_004: [ wake_by_address_all shall cause all the thread(s) waiting on a call to wait_on_address with argument address to continue execution. ]*/
+/*Codes_SRS_SYNC_43_005: [ wake_by_address_single shall cause one thread waiting on a call to wait_on_address with argument address to continue execution. ]*/
+TEST_FUNCTION(wake_up_all_threads)
+{
+    ///arrange
+    volatile_atomic int32_t var;
+    interlocked_exchange(&var, 0);
+    interlocked_exchange(&create_count, 0);
+    THREAD_HANDLE threads[100];
+
+    ///act
+    for(int i = 0; i < 100; ++i){
```

Comment:
>for(int i = 0; i < 100; ++i){ [](start = 4, length = 29)

style: curly brace goes on new line by itself, space between for and opening parenthesis.